apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -5,6 +5,11 @@ class ProjectsController < ApplicationController
   # GET /projects/:id
   def show
     @project = Project.find(params[:id])
+    if @project.nil?
+      redirect_to(projects_path, alert: "cannot find project #{params[:id]}")
+    else
+      @scripts = Script.all(@project.directory)
+    end
```

Comment:
If the project is not `nil` then return an array of of the `scripts`? Is that the idea of this controller patch?

Reply:
If the projects is nil, then we redirect to the index with an alert.

Reply:
Yes, I'm asking about the `script` array though. Why is this being handed back if it is _not_ `nil`?

Reply:
:facepalm: I see now. Sorry for the confusion.